Libraries/builtin/Sources/copy/Driver.cpp

```diff
@@ -39,33 +39,48 @@ name()
 }
 
 static bool
-CopyPath(
-    Filesystem *filesystem,
-    std::unordered_map<std::string, std::string> const &environmentVariables,
-    std::string const &workingDirectory,
-    std::string const &inputPath,
-    std::string const &outputPath)
+CopyPath(Filesystem *filesystem, std::string const &inputPath, std::string const &outputPath)
 {
     if (!filesystem->createDirectory(FSUtil::GetDirectoryName(outputPath))) {
         return false;
     }
 
-    process::Subprocess cp;
-    if (!cp.execute(filesystem, "/bin/cp", { "-R", inputPath, outputPath }, environmentVariables, workingDirectory) || cp.exitcode() != 0) {
+    // TODO: This shouldn't use external tools to copy.
+    process::Context const *context = process::Context::GetDefaultUNSAFE();
+    process::Launcher *launcher = process::Launcher::GetDefaultUNSAFE();
+
+    process::MemoryContext cp = process::MemoryContext(
+        "/bin/cp",
+        context->currentDirectory(),
+        { "-R", inputPath, outputPath },
+        context->environmentVariables(),
```

Comment:
I originally had this code doing that with setters:

   process::MemoryContext context = process::MemoryContext(processContext);
   context.executablePath() = "/bin/cp";
   context.commandLineArguments() = { "-R", inputPath, outputPath };

But that seemed too easy to forget some options. This way is explicit about which are copied and which are specified.